src/wallet3/transaction_constructor.cpp

```diff
@@ -29,6 +31,179 @@ namespace wallet
     return new_tx;
   }
 
+  PendingTransaction
+  TransactionConstructor::create_ons_buy_transaction(
+      const cryptonote::tx_destination_entry& change_recipient,
+      const std::string& type_str,
+      const std::string& owner_str,
+      const std::string& backup_owner_str,
+      const std::string& name,
+      const std::string& value
+      )
+  {
+    std::vector<cryptonote::tx_destination_entry> recipients;
+    PendingTransaction new_tx(recipients);
+    auto [hf, hf_uint8] = cryptonote::get_ideal_block_version(db->network_type(), db->scan_target_height());
+    cryptonote::oxen_construct_tx_params tx_params{hf, cryptonote::txtype::oxen_name_system, 0, 0};
```

Comment:
Is this variable needed?  It's referenced twice, just (apparently) to get out the values that we are passing into the constructor.